diff --git a/drivers/android/Kconfig b/drivers/android/Kconfig
index 25ed16b68..6f8aad691 100644
--- a/drivers/android/Kconfig
+++ b/drivers/android/Kconfig
@@ -62,7 +62,7 @@ config BINDER_OPT
 
 config ANDROID_SIMPLE_LMK
 	bool "Simple Android Low Memory Killer"
-	depends on !ANDROID_LOW_MEMORY_KILLER && !MEMCG && !PSI
+	depends on !ANDROID_LOW_MEMORY_KILLER && !PSI
 	---help---
 	  This is a complete low memory killer solution for Android that is
 	  small and simple. Processes are killed according to the priorities
