diff --git a/drivers/misc/Kconfig b/drivers/misc/Kconfig
index 796561ccc..bd004ab50 100644
--- a/drivers/misc/Kconfig
+++ b/drivers/misc/Kconfig
@@ -4,6 +4,17 @@
 
 menu "Misc devices"
 
+config NTSYNC
+        tristate "NT synchronization primitive emulation"
+        help
+          This module provides kernel support for emulation of Windows NT
+          synchronization primitives. It is not a hardware driver.
+
+          To compile this driver as a module, choose M here: the
+          module will be called ntsync.
+
+          If unsure, say N.
+
 config SPI_XIAOMI_TP
         bool
         depends on SPI
diff --git a/drivers/misc/Makefile b/drivers/misc/Makefile
index 65da65cc0..51cda9f44 100644
--- a/drivers/misc/Makefile
+++ b/drivers/misc/Makefile
@@ -2,7 +2,7 @@
 #
 # Makefile for misc devices that really don't fit anywhere else.
 #
-
+obj-$(CONFIG_NTSYNC)            += ntsync.o
 obj-$(CONFIG_IBM_ASM)		+= ibmasm/
 obj-$(CONFIG_IBMVMC)		+= ibmvmc.o
 obj-$(CONFIG_AD525X_DPOT)	+= ad525x_dpot.o
